fdbserver/DDTxnProcessor.actor.cpp

```diff
@@ -538,3 +538,168 @@ Future<HealthMetrics> DDTxnProcessor::getHealthMetrics(bool detailed) const {
+		KeyRangeRef curRange = it->range();
+		DDShardInfo info(curRange.begin);
+
+		auto teams = mgs->shardMapping->getTeamsFor(curRange);
+		if (!teams.first.empty() && !teams.second.empty()) {
+			CODE_PROBE(true, "Mock InitialDataDistribution In-Flight shard");
+			info.hasDest = true;
+			info.destId = anonymousShardId;
+			info.srcId = anonymousShardId;
+			transformTeamsToServerIds(teams.second, info.primarySrc, info.remoteSrc);
+			transformTeamsToServerIds(teams.first, info.primaryDest, info.remoteDest);
+		} else if (!teams.first.empty()) {
+			CODE_PROBE(true, "Mock InitialDataDistribution Static shard");
+			info.srcId = anonymousShardId;
+			transformTeamsToServerIds(teams.first, info.primarySrc, info.remoteSrc);
+		} else {
+			ASSERT(false);
+		}
+
+		res.push_back(std::move(info));
+	}
+	res.emplace_back(allKeys.end);
+
+	return res;
+}
+
+Future<Reference<InitialDataDistribution>> DDMockTxnProcessor::getInitialDataDistribution(
+    const UID& distributorId,
+    const MoveKeysLock& moveKeysLock,
+    const std::vector<Optional<Key>>& remoteDcIds,
+    const DDEnabledState* ddEnabledState) {
+
+	// FIXME: now we just ignore ddEnabledState and moveKeysLock, will fix it in the future
+	Reference<InitialDataDistribution> res = makeReference<InitialDataDistribution>();
+	res->mode = 1;
+	res->allServers = getServerListAndProcessClasses().get();
+	res->shards = getDDShardInfos();
+	std::tie(res->primaryTeams, res->remoteTeams) = getAllTeamsInRegion(res->shards);
+	return res;
+}
+
+Future<Void> DDMockTxnProcessor::removeKeysFromFailedServer(const UID& serverID,
+                                                            const std::vector<UID>& teamForDroppedRange,
+                                                            const MoveKeysLock& lock,
+                                                            const DDEnabledState* ddEnabledState) const {
+
+	// This function only takes effect when user exclude failed IP:PORT in the fdbcli. In the first version , the mock
+	// class won’t support this.
+	UNREACHABLE();
+}
+
+Future<Void> DDMockTxnProcessor::removeStorageServer(const UID& serverID,
+                                                     const Optional<UID>& tssPairID,
+                                                     const MoveKeysLock& lock,
+                                                     const DDEnabledState* ddEnabledState) const {
+	ASSERT(mgs->allShardRemovedFromServer(serverID));
+	mgs->allServers.erase(serverID);
+	return Void();
+}
+
+void DDMockTxnProcessor::setupMockGlobalState(Reference<InitialDataDistribution> initData) {
+	for (auto& [ssi, pInfo] : initData->allServers) {
+		mgs->addStorageServer(ssi);
+	}
+	mgs->shardMapping->setCheckMode(ShardsAffectedByTeamFailure::CheckMode::ForceNoCheck);
+
+	for (int i = 0; i < initData->shards.size() - 1; ++i) {
+		// insert to keyServers
+		auto& shardInfo = initData->shards[i];
+		ASSERT(shardInfo.remoteSrc.empty() && shardInfo.remoteDest.empty());
```

Comment:
Is this something to be added later?

Reply:
The `remoteSrc`and `remoteDest` are only used in multi-region setting. Currently, I just make all MGS tests disable the remote region configuration by using `fearless=false` in the test spec files.
In the short period, I'm not planning to support multi-region tests in MGS. This will be change when we find time to add region support in MGS